pytorch_lightning/utilities/device_parser.py

```diff
@@ -184,11 +184,13 @@ def _normalize_parse_gpu_input_to_list(gpus: Union[int, List[int], Tuple[int, ..
     return list(range(gpus))
 
 
-def _get_all_available_gpus() -> List[int]:
+def _get_all_available_gpus() -> Optional[List[int]]:
     """
     Returns:
          a list of all available gpus
     """
+    if not torch.cuda.is_available():
+        return None
```

Comment:
Hey @Borda, explicitly asking for gpus and silently not getting them when on a CPU-only machine may be problematic. What do you say?

Reply:
it is not silently, you just say give me the best, so if you have GPU, give me all, if you run the same on a notebook, do not crash, as it is not with `-1` I can't run the same code anywhere a need to change it...

Reply:
@awaelchli this is the usecase in Flash, I wanted to use `-1` but it would crash so I have to use `torch.cuda.device_count()`, https://github.com/PyTorchLightning/lightning-flash/pull/638

Reply:
Actually I think it's ok since we print GPU available: True/False GPU used: True/False with every lIghtning training. 
so adding a warning would not do much more. 

don't forget to update tests :)